main/eval/src/mill/eval/EvaluatorCore.scala

```diff
@@ -180,8 +180,13 @@ private[mill] trait EvaluatorCore extends GroupEvaluator {
     val (_, tasksTransitive0) = Plan.plan(Agg.from(tasks0.map(_.task)))
 
     val tasksTransitive = tasksTransitive0.toSet
-    val (tasks, leafCommands) = terminals0.partition {
-      case Terminal.Labelled(t, _) if tasksTransitive.contains(t) => true
+    val (tasks, leafSerialCommands) = terminals0.partition {
+      case Terminal.Labelled(t, _) =>
+        if (tasksTransitive.contains(t)) true
+        else t match {
+          case t: Command[_] => !t.exclusive
+          case _ => false
```

Comment:
Shouldn't this be a `true` or a `!serialCommandExec`? 

Maybe, `serialCommandExec` isn't need anymore, since we now have fine control over exclusive execution.